### PR TITLE
Ensure we are installing with / running on Python 3

### PR DESCRIPTION
**Pull request by stefanv**
Closes gh-3117

### CODE REVIEW

**Comment by sciunto** [comment]
LGTM.

**Comment by stefanv** [comment]
@sciunto You have merge privileges. Go for it!

**Comment by sciunto** [comment]
Bang! :)

File: setup.py [hunk 1]
```diff
@@ -25,10 +25,21 @@
 from distutils.command.build_py import build_py
 from distutils.command.sdist import sdist
 
-if sys.version_info[0] < 3:
-    import __builtin__ as builtins
-else:
-    import builtins
+if sys.version_info < (3, 4):
```
Comment (soupault) [inline setup.py]: Shouldn't this be (3, 5)?

File: setup.py [hunk 2]
```diff
@@ -25,10 +25,21 @@
...
+
+    error = """Python {py} detected.
+
+scikit-image 0.15+ support only Python 3.5 and above.
```
Comment (soupault) [inline setup.py]: Or 3.4 here?

**Comment by stefanv** [comment]
Say da da da da.

I wonder who can decode that one :)

**Comment by stefanv** [comment]
Yes, probably! You can go ahead and push that fix to master.

**Comment by soupault** [comment]
Having Python 3.4 EOL in 2019 (even before Python 2.7, https://devguide.python.org/#status-of-python-branches), the comparison above should be to 3.5.
Fixed in https://github.com/scikit-image/scikit-image/commit/1b379580782180f4e506864949b6257d9c43a94c.

**Comment by jni** [comment]
Thanks @soupault! 3.4 was never the right choice. I might even argue for 3.6, but this debate can wait a bit. Thank you for the push and best of luck with your move!

**Pull request merged**